Sources/Vapor/Sessions/Request+Session.swift

```
@@ -9,38 +9,45 @@ extension Request {
     ///
     /// - note: `SessionsMiddleware` must be added and enabled.
     /// - returns: `Session` for this `Request`.
-    public func session() throws -> Session {
-        guard let cache = self._sessionCache else {
-            fatalError("No session cache.")
-        }
-        if !cache.middlewareFlag {
+    public var session: Session {
+        if !self._sessionCache.middlewareFlag {
             // No `SessionsMiddleware` was detected on your app.
             // Suggested solutions:
             // - Add the `SessionsMiddleware` globally to your app using `MiddlewareConfig`
             // - Add the `SessionsMiddleware` to a route group.
             assertionFailure("No `SessionsMiddleware` detected.")
```

Could this be a vapor error where you get the suggestions and stuff in your console? Or some other kind of logging?

---

@MrLotU that would require that `session` being throwing, which makes a whole bunch of other methods throwing, too. Given that this error is a development-time error, I think it makes sense to fail hard and fast.

---

Makes sense, maybe log a few lines before crashing though?
```swift
logger.log("""
No `SessionsMiddleware` was detected on your app.
Suggested solutions:
- Add the `SessionsMiddleware` globally to your app using `MiddlewareConfig`
- Add the `SessionsMiddleware` to a route group.
""")
assertionFailure("No `SessionsMiddleware` detected.")
```